src/libopenrave-core/jsonparser/jsonreader.cpp

```diff
@@ -384,45 +451,52 @@ class JSONReader
         return false;
     }
 
-    boost::shared_ptr<const rapidjson::Document> _GetDocumentFromFilename(const std::string& fullFilename, rapidjson::Document::AllocatorType& alloc)
+    boost::shared_ptr<const rapidjson::Document> _GetDocumentFromFilename(const std::string& referenceURI)
     {
         boost::shared_ptr<const rapidjson::Document> doc;
+
         // TODO: optimize this. for the first time doc is cached, all the expandable object will never get cached, because we are not update document cache after expand any body
-        if (_rapidJSONDocuments.find(fullFilename) != _rapidJSONDocuments.end()) {
-            doc = _rapidJSONDocuments[fullFilename];
+
+        if (_rapidJSONDocuments.find(referenceURI) != _rapidJSONDocuments.end()) {
+            doc = _rapidJSONDocuments[referenceURI];
         }
         else {
-            boost::shared_ptr<rapidjson::Document> newDoc;
-            if (_EndsWith(fullFilename, ".json")) {
-                newDoc.reset(new rapidjson::Document(&alloc));
-                OpenRapidJsonDocument(fullFilename, *newDoc);
+            boost::shared_ptr<rapidjson::Document> newDoc = boost::shared_ptr<rapidjson::Document>(new rapidjson::Document());;
+            if (_EndsWith(referenceURI, ".json") && !_StartsWith(referenceURI, "http")) {
```

Comment:
This is not a good check. Reason already mentioned above.

Reply:
removed starts with